var/spack/repos/builtin/packages/gzip/package.py

```diff
@@ -13,3 +14,8 @@ class Gzip(AutotoolsPackage):
     url      = "https://ftp.gnu.org/gnu/gzip/gzip-1.10.tar.gz"
 
     version('1.10', sha256='c91f74430bf7bc20402e1f657d0b252cb80aa66ba333a25704512af346633c68')
+
+    @property
+    def build_directory(self):
+        # Gzip makes a recursive symlink if built in-source
+        return os.path.join(self.stage.path, 'spack-build')
```

Comment:
You can replace this whole thing with:
```python
build_directory = 'spack-build'
```
I tested this on my macOS laptop and it worked. Also, can you add a build dependency on `gmake` while you're at it? Only `gmake` supports `GNUmakefile`, the other `make` implementations don't.

Reply:
Done.